src/main/java/com/rabbitmq/client/impl/AMQConnection.java

```diff
@@ -245,10 +252,17 @@ public AMQConnection(ConnectionParams params, FrameHandler frameHandler, Metrics
         this._inConnectionNegotiation = true; // we start out waiting for the first protocol response
 
         this.metricsCollector = metricsCollector;
+
+        this.errorOnWriteListener = params.getErrorOnWriteListener() != null ? params.getErrorOnWriteListener() :
+            new ErrorOnWriteListener() {
+                @Override
+                public void handle(Connection connection, IOException exception) { }
```

Comment:
to maintain passivity shouldn't this rethrow the exception instead of swallow it?

Reply:
Can you clarify what you mean by "passivity" here? The goal of this listener is to kick off connection recovery first and foremost. We do not want to re-throw anything in that case as there won't be anything to catch it.

Reply:
Yes, this should re-throw the exception as-is in this case (hence passivity). This applies only to non-recoverable connections. The [listener is called in the `flush` method](https://github.com/rabbitmq/rabbitmq-java-client/blob/e934ed31156731965bcb68c90e2382db5d64876d/src/main/java/com/rabbitmq/client/impl/AMQConnection.java#L569-L575) and at first the exception was always re-thrown after the call to the listener. I then switched the exception handling in the listener to give it more control but forgot to reflect it in this default, no-op implementation. Again, this impacts only non-recoverable connections (when the user disables automatic connection recovery explicitly), `AutorecoveringConnection` [sets up a listener that triggers connection recovery](https://github.com/rabbitmq/rabbitmq-java-client/blob/e934ed31156731965bcb68c90e2382db5d64876d/src/main/java/com/rabbitmq/client/impl/recovery/AutorecoveringConnection.java#L103-L126). Thanks for pointing this out @vikinghawk!